lib/iris/cube.py

```diff
@@ -912,6 +927,38 @@ def add_aux_factory(self, aux_factory):
                             'iris.aux_factory.AuxCoordFactory.')
         self._aux_factories.append(aux_factory)
 
+    def add_cell_measure(self, cell_measure, data_dims=None):
+        """
+        Adds a CF cell measure to the cube.
+
+        Args:
+
+        * cell_meaasure
+            The :class:`iris.coords.CellMeasure`
+            instance to add to the cube.
+
+        Kwargs:
+
+        * data_dims
+            Integer or iterable of integers giving the data dimensions spanned
+            by the coordinate.
+
+        Raises a ValueError if a cell_measure with identical metadata already
+        exists on the cube.
+
+        See also
+        :meth:`Cube.remove_cell_measure()<iris.cube.Cube.remove_cell_measure>`.
+
+        """
+        if self.cell_measures(cell_measure):
+            raise ValueError('Duplicate cell_measures are not permitted.')
+        self._add_unique_cell_measure(cell_measure, data_dims)
+
+    def _add_unique_cell_measure(self, cell_measure, data_dims):
```

Comment:
@marqh Is there a reason why you've created this private method ?

I can't find reference to its usage anywhere else in the code base other than from `Cube.add_cell_measure`.

Most likely it's existence is lingering from a previous implementation ... if so, could we blend this functionality directly into `Cube.add_cell_measure` ?